homeassistant/components/devolo_home_network/system_health.py

```diff
@@ -0,0 +1,40 @@
+"""Provide info to system health."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from devolo_plc_api.exceptions.device import DeviceUnavailable
+
+from homeassistant.components import system_health
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+
+
+@callback
+def async_register(
+    hass: HomeAssistant, register: system_health.SystemHealthRegistration
+) -> None:
+    """Register system health callbacks."""
+    register.async_register_info(system_health_info)
+
+
+async def system_health_info(hass: HomeAssistant) -> dict[str, Any]:
+    """Get info for the info page."""
+    states: list[bool] = []
+    firmware_updates_available = await asyncio.gather(
+        *[
+            entity["device"].device.async_check_firmware_available()
+            for entity in hass.data[DOMAIN].values()
+        ],
```

Comment:
Firmware update should be exposed as binary sensors with the `update` device class.

Reply:
I see. So then this MR does not make sence as it is. I will close it.